test/helpers.go

```
@@ -2719,6 +2719,42 @@ type ControllerLogSummary struct {
 
+// CheckPodsForImagePullErrors checks if any pods in the given namespace have ErrImagePull or
+// ImagePullBackOff status. Returns an error describing the affected pods if found, nil otherwise.
+func CheckPodsForImagePullErrors(t *testing.T, kubeContext, namespace string) error {
+	t.Helper()
+
+	output, err := RunCommandQuiet(t, "kubectl", "--context", kubeContext,
+		"-n", namespace, "--request-timeout=10s",
+		"get", "pods",
+		"-o", "jsonpath={range .items[*]}{.metadata.name}{\"\\t\"}{range .status.containerStatuses[*]}{.state.waiting.reason}{\" \"}{end}{\"\\n\"}{end}")
+	if err != nil {
+		return nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 2727) but it returns nil

(nilerr)